File: game/src/screen/Screen.js

var Screen = function() 
{   this.game = null;   
};

Screen.prototype.$ = function(game)
{
    this.game = game;
    return this;
};

Screen.prototype.discard = function()
{
};

Screen.prototype.setDirty = function()
{
    this.game.setDirty();
};


// can be overwritten to get notified about events and things to do
// and things to report


Screen.prototype.tick = function()
{
};

Screen.prototype.draw = function()
{
};

Screen.prototype.reactivate = function()
{
};

Screen.prototype.isOverlay = function()
{   return false;
};

Screen.prototype.onResize = function()
{
};

// ---- default key event handlers  ----
Screen.prototype.onKeyDown = function(keycode)
{
};

Screen.prototype.onKeyUp = function(keycode)
{
};

Screen.prototype.onBackNavigation = function()
{
    this.game.removeScreen();
};
    
// interface for simplified touch events. 
Screen.prototype.onPointerDown = function(x, y)
{
};

Screen.prototype.onPointerUp = function()
{
};

Screen.prototype.onPointerMove = function(x,y)
{
};

File: SapphireYours_Common/src/grafl/sy/screens/Screen.java
package grafl.sy.screens;
 

import grafl.sy.game.Game;
import android.view.KeyEvent;

public abstract class Screen 
{
	public final Game game;	
	
	public int screenwidth;
	public int screenheight;
	
	public Screen(Game game)
	{
		this.game = game;
		screenwidth = 0;
		screenheight = 0;
	}
	
	public void resize(int width, int height)
	{
		screenwidth = width;
		screenheight = height;
	}
	
	public void discard()
	{
	}
	
	public void tick()
	{
	}
	
	public void draw()
	{
	}
		
				
	public void reactivate()
	{
	}
	
	public boolean isOverlay()
	{	return false;
	}
		
	
    // ---- key event handlers called in GL thread ----
    public void handleKeyEvent(KeyEvent event)
    {
    }
	
	public void handleBackNavigation()
	{
		game.removeScreen();	
	} 	
    
    /** interface for simplified touch events. 
     */       
	public void onPointerDown(int x, int y)
	{
	}
	public void onPointerUp()
	{
	}
	public void onPointerMove(int x, int y)
	{
	}


	// some toolbox methods to be used on various screens
	
	/** 
	 * 	Create string of the form m:ss of a given number of seconds. 
	 *  Only non-negative seconds work correctly.
	 */ 
	public static String buildTimeString(int seconds)
	{
		int m = Math.floor(seconds/60);
		int s = seconds%60;
        if (s>=10)
        {	return m+":"+s;
        }
        else
        {	return m+":0"+s;
        }			
	} 
	
}
